pkg/controller/tidbcluster/tidb_cluster_control.go

```diff
@@ -114,6 +115,8 @@ func (tcc *defaultTidbClusterControl) validate(tc *v1alpha1.TidbCluster) bool {
 }
 
 func (tcc *defaultTidbClusterControl) updateTidbCluster(tc *v1alpha1.TidbCluster) error {
+	// default the tidbcluster
+	defaulting.SetTidbClusterDefault(tc)
```

Comment:
move it to this function, we can have our controller-side validation and defaulting at the same place

https://github.com/pingcap/tidb-operator/blob/52491834b6e0371a4ae9bd862401f4d398d27e61/pkg/controller/tidbcluster/tidb_cluster_control.go#L84-L87

Reply:
updated.